app/src/main/java/org/fossasia/openevent/app/events/EventDetailActivity.java

```diff
@@ -0,0 +1,90 @@
+
+import org.fossasia.openevent.app.R;
+import org.fossasia.openevent.app.data.models.Event;
+import org.fossasia.openevent.app.event.EventContainerFragment;
+import org.fossasia.openevent.app.listeners.OnFragmentInteractionListener;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+/**
+ * An activity representing a single Event detail screen. This
+ * activity is only used narrow width devices. On tablet-size devices,
+ * item details are presented side-by-side with a list of items
+ * in a {@link EventListActivity}.
+ */
+public class EventDetailActivity extends AppCompatActivity implements OnFragmentInteractionListener {
+
+    @BindView(R.id.detail_toolbar)
+    Toolbar toolbar;
+
+    private Event initialEvent;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_event_detail);
+
+        ButterKnife.bind(this);
+
+        setSupportActionBar(toolbar);
+
+        // Show the Up button in the action bar.
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        initialEvent = getIntent().getParcelableExtra(EventListActivity.EVENT_KEY);
+
+        setTitle(initialEvent.getName());
+
+        // savedInstanceState is non-null when there is fragment state
+        // saved from previous configurations of this activity
+        // (e.g. when rotating the screen from portrait to landscape).
+        // In this case, the fragment will automatically be re-added
+        // to its container so we don't need to manually add it.
+        // For more information, see the Fragments API guide at:
+        //
+        // http://developer.android.com/guide/components/fragments.html
+        //
+        if (savedInstanceState == null) {
+            // Create the detail fragment and add it to the activity
+            // using a fragment transaction.
+            EventContainerFragment fg = EventContainerFragment.newInstance(initialEvent);
+            getSupportFragmentManager().beginTransaction()
+                .add(R.id.event_detail_container, fg)
+                .commit();
+        }
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
+        if (id == android.R.id.home) {
+            // This ID represents the Home or Up button. In the case of this
+            // activity, the Up button is shown. For
+            // more details, see the Navigation pattern on Android Design:
+            //
+            // http://developer.android.com/design/patterns/navigation.html#up-vs-back
+            //
+            navigateUpTo(new Intent(this, EventListActivity.class));
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onFragmentInteraction(Uri uri) {
```

Comment:
Yeah I will remove it. It is just that till now we didn't require any communication from fragment to the host.